.github/workflows/push-chart-releases.yaml

```
@@ -0,0 +1,95 @@
+name: Chart Release Push
+
+on:
+  # run after the image build completes
+  workflow_run:
+    workflows:
+      - Image Release Build
+    types:
+      - completed
+  # allow manually triggering it as well, for existing tags
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: "Git tag to release - with the v prefix"
+        required: true
+
+permissions:
+  # check out source code
+  contents: read
+  # fetch artifact digests
+  actions: read
+  # sign helm chart
+  id-token: write
+
+jobs:
+  push-chart:
+    name: Push Chart
+    runs-on: ubuntu-24.04
+    if: ${{ github.event.workflow_run.conclusion == 'success' || github.event_name == 'workflow_dispatch' }}
+
```

For release artifacts we will need:
```suggestion
  push-chart:
    name: Push Chart
    runs-on: ubuntu-24.04
    if: ${{ github.event.workflow_run.conclusion == 'success' || github.event_name == 'workflow_dispatch' }}
    environment: release
```

@aanm do you think it would be better to integrate this step into..

1. https://github.com/cilium/cilium/blob/main/.github/workflows/build-images-releases.yaml
2. https://github.com/cilium/cilium/blob/main/.github/workflows/release.yaml